src/plugin/pelias/geocoder/search.js

```diff
@@ -47,32 +47,49 @@ goog.inherits(plugin.mapzen.places.Search, os.search.AbstractUrlSearch);
  * @private
  * @const
  */
-plugin.mapzen.places.Search.LOGGER_ = goog.log.getLogger('plugin.mapzen.places.Search');
+plugin.pelias.geocoder.Search.LOGGER_ = goog.log.getLogger('plugin.pelias.geocoder.Search');
 
 
 /**
  * @inheritDoc
  */
-plugin.mapzen.places.Search.prototype.getSearchUrl = function(term, opt_start, opt_pageSize) {
-  var url = /** @type {?string} */ (os.settings.get(['plugin', 'mapzen', 'places', 'url']));
+plugin.pelias.geocoder.Search.prototype.getSearchUrl = function(term, opt_start, opt_pageSize) {
+  var url = /** @type {?string} */ (os.settings.get(['plugin', 'pelias', 'geocoder', 'url']));
 
-  var boundary = /** @type {string} */ (os.settings.get(['plugin', 'mapzen', 'places', 'extentParams']));
+  var boundary = /** @type {boolean} */ (os.settings.get(['plugin', 'pelias', 'geocoder', 'extentParams']));
 
   if (boundary) {
     // if the view is small enough, we'll apply a bounding rectangle to the search
     // defaults to 200km
-    var threshold = /** @type {number} */ (os.settings.get(['plugin', 'mapzen', 'places', 'extentThreshold'], 200000));
+    var threshold =
+        /** @type {number} */ (os.settings.get(['plugin', 'pelias', 'geocoder', 'extentThreshold'], 200000));
     var extent = os.MapContainer.getInstance().getMap().getExtent();
 
     // translate to lon/lat
     extent = ol.proj.transformExtent(extent, os.map.PROJECTION, os.proj.EPSG4326);
     var distance = osasm.geodesicInverse(extent.slice(0, 2), extent.slice(2, 4)).distance;
 
     if (distance <= threshold) {
-      url += boundary;
+      var boundaryText = '&boundary.rect.min_lat=' + extent[1]
+        + '&boundary.rect.min_lon=' + extent[0]
+        + '&boundary.rect.max_lat=' + extent[3]
+        + '&boundary.rect.max_lon=' + extent[2];
+      url += boundaryText;
     }
   }
 
+  // Add the focus.point modifiers to the URL if enabled and we are zoomed in to at least zoom level 4.
+  var focuspoint = /** @type {boolean} */ (os.settings.get(['plugin', 'pelias', 'geocoder', 'focusPoint']));
+  if (focuspoint) {
+    var threshold = /** @type {number} */ (os.settings.get(['plugin', 'pelias', 'geocoder', 'focusPointMinZoom'], 4.0));
+    var currentZoom = /** @type {number} */ (os.MapContainer.getInstance().getMap().getView().getZoom());
+    if (currentZoom >= threshold) {
+      var centre = /** @type {Array<number>} */ (os.MapContainer.getInstance().getMap().getView().getCenter());
```

Comment:
And also here?

Reply:
This one is needed:
```
/home/bradh/opensphere-yarn-workspace/workspace/opensphere/src/plugin/pelias/geocoder/search.js:88: ERROR - actual parameter 1 of ol.proj.toLonLat does not match formal parameter
found   : (Array<number>|null|undefined)
required: (Array<number>|null)
      var centreLonLat = ol.proj.toLonLat(centre, os.map.PROJECTION);
                                          ^^^^^^
```

Reply:
I'd recommend this instead, as the cast doesn't actually prevent an error:
```
var center = os.MapContainer.getInstance().getMap().getView().getCenter();
if (center) {
 ...
}
```

In other words, if the coordinate can actually be `null` or `undefined`, the cast only satisfies the compiler (you just soothe it with a little white lie). The app itself can still return those values and break your code (the `null` or `undefined` coordinate case is more common in the 3D view).

Reply:
Thanks. Found another problem in the `getExtent` call above too (where the 2D map returns > 180 or < -180). Maybe can use the view there too.